### ISSUE TITLE
[TensorFlow 2.0] Automated naming of keras metric layers is not consistent

### ISSUE THREAD

**Issue by xlambein** [tensorflow/tensorflow#30310]
**System information**
- Have I written custom code: **yes**
- OS Platform and Distribution: **Mac OS X 10.14.5**
- TensorFlow installed from: **binary**
- TensorFlow version: **v2.0.0-beta0-16-g1d91213fe7 2.0.0-beta1**
- Python version: **3.6.8**
- CUDA/cuDNN version: **don't have a GPU**
- GPU model and memory: **don't have a GPU**

**Describe the current behavior**

The automatic naming of Keras metric layers is not consistent.  Example, if I create two `BinaryAccuracy` metrics:

```python
>>> tf.keras.metrics.BinaryAccuracy().name
'binary_accuracy'
>>> tf.keras.metrics.BinaryAccuracy().name
'binary_accuracy'
```

Whereas, if I create two `Recall` metrics:

```python
>>> tf.keras.metrics.Recall().name
'recall'
>>> tf.keras.metrics.Recall().name
'recall_1'
```

After testing every metric class in `tf.keras.metrics`, I found that the same behavior as `Recall` is produced by the following metrics:

- `AUC`
- `FalseNegatives`
- `FalsePositives`
- `MeanIoU`
- `MeanRelativeError`
- `Precision`
- `Recall`
- `SensitivityAtSpecificity`
- `SpecificityAtSensitivity`
- `TrueNegatives`
- `TruePositives`

**Describe the expected behavior**

Either all the metrics should be created with the exact same name, or they should all have an integer added to the end of the second metric with that name.

**Code to reproduce the issue**

Here's my code that lists all the classes in `tf.keras.metrics` that have different names when instanciated twice:

```python
import tensorflow as tf
import inspect

for name, metric in tf.keras.metrics.__dict__.items():
    if inspect.isclass(metric):  # It's a metric keras layer
        args = ()
        if name == 'MeanIoU':
            args = (2,)
        elif name == 'MeanRelativeError':
            args = ([1],)
        elif name == 'Metric':
            continue
        elif name == 'SensitivityAtSpecificity':
            args = (0.5,)
        elif name == 'SpecificityAtSensitivity':
            args = (0.5,)

        layer_name_0 = metric(*args).name
        layer_name_1 = metric(*args).name

        if layer_name_0 != layer_name_1:  # Two same metrics don't have the same name
            print('-', name)
```

**Comment by gadagashwini** [comment]
I am able to reproduce the issue on Colab with Tensorflow 2.0.0.beta1. Thanks!

**Comment by ymodak** [comment]
I think this is intended behavior.  Referring to says ```tf.keras.metrics.Recall``` documentation [```Creates a Recall instance.```](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Recall#__init__ ).
Whereas for ```tf.keras.metrics.Accuracy``` the name: Accuracy is hard coded.
Same can apply to the other metrics as well.

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
